api/handlers.go

```diff
@@ -265,7 +265,7 @@ func diagnosticsJobStatusAllHandler(w http.ResponseWriter, r *http.Request) {
 func cancelBundleReportHandler(w http.ResponseWriter, r *http.Request) {
 	dt, ok := getDtFromContext(r.Context())
 	if !ok {
-		httpError(w, "Unable to get a required context from a request", http.StatusInternalServerError)
+		httpError(w, "Unable to get a required context from a request", http.StatusBadRequest)
```

Comment:
Can we make these errors more descriptive please? I won't block the PR on it but it'd be really nice :)

Reply:
Sure I have one PR to fix config and fixing this error codes will be next. I'll also add tests for it. My goal is to get health coverage (about 70%) and then start changing the behaviour.